crates/compose/src/lib.rs

```diff
@@ -42,6 +43,36 @@ pub trait ComponentSourceLoader {
     ) -> anyhow::Result<Vec<u8>>;
 }
 
+/// A ComponentSourceLoader that loads component sources from the filesystem.
+pub struct ComponentSourceLoaderFs;
+
+#[async_trait::async_trait]
+impl ComponentSourceLoader for ComponentSourceLoaderFs {
```

Comment:
Why did this structure move to compose from loader?

Reply:
To share the loader between the trigger crate and the oci; it's a fairly generic impl that isn't tied to anything trigger specific.